crawl-ref/source/ability.cc

```diff
@@ -1826,6 +1829,17 @@ static spret_type _do_ability(const ability_def& abil, bool fail)
         }
         return frog_hop(fail);
 
+    case ABIL_SHIFT_ATTRIBUTES:
+    {
+        if (you.attribute[ATTR_XP_DRAIN] > 0)
```

Comment:
Removed the check as suggested.  It was needed in the original implementation of the ability to prevent abuse (shifting to Int to cast a spell, then shifting back immediately afterward), but the current drain cost should be sufficient enough to prevent that from happening.